samples/bluetooth/peripheral_fast_pair/src/bt_adv_helper.c

```diff
@@ -149,20 +149,20 @@ static int adv_start_internal(enum bt_fast_pair_adv_mode fp_adv_mode)
 
 static void rpa_rotate_fn(struct k_work *w)
 {
-	(void)adv_start_internal(adv_helper_fp_adv_mode);
+	(void)adv_start_internal(adv_helper_fp_adv_config);
 }
 
-int bt_adv_helper_adv_start(enum bt_fast_pair_adv_mode fp_adv_mode)
+int bt_adv_helper_adv_start(struct bt_fast_pair_adv_config fp_adv_config)
```

Comment:
I wonder if we actually need to pass whole advertising configuration here. We should decide on how we want to integrate updating battery values and UI indication toggling to the sample (e.g. I think that RPA cannot be updated every time a battery value changes)

Reply:
Maybe handling battery values could be done in a separate source file (new Fast Pair advertising data provider API allows us to even set battery notification configuration separately). The file would also be responsible for GATT Battery Service updates (misalignment between battery values in notification and BAS could be confusing I think).